### PR TITLE
Remove Python 3.7 support

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: CI.rst [hunk 1]
```diff
@@ -145,7 +145,7 @@ have to be percent-encoded when you access them via UI (/ = %2F)
 +--------------+----------------------------------------------------------+----------------------------------------------------------+
 
 * <BRANCH> might be either "main" or "v2-*-test"
-* <X.Y> - Python version (Major + Minor).Should be one of ["3.7", "3.8", "3.9"].
+* <X.Y> - Python version (Major + Minor).Should be one of ["3.8", "3.9", "3.10"].
```
Comment (uranusjr) [inline CI.rst]: Should this include 3.11 now?

File: airflow/providers/alibaba/cloud/log/oss_task_handler.py [hunk 2]
```diff
@@ -21,10 +21,10 @@
 import os
 import pathlib
 import shutil
+from functools import cached_property
 
 from packaging.version import Version
 
-from airflow.compat.functools import cached_property
```
Comment (uranusjr) [inline airflow/providers/alibaba/cloud/log/oss_task_handler.py:27]: Can we do this to providers just yet?

File: CI.rst (hunk 1 above)
Comment (potiuk) [reply CI.rst]: Yep. It's a draft i keep on rebasing and fixing basic issues incrementallly - so those things will have to be reviewed and updated

File: airflow/providers/alibaba/cloud/log/oss_task_handler.py (hunk 2 above)
Comment (potiuk) [reply airflow/providers/alibaba/cloud/log/oss_task_handler.py:27]: That is a good point. I think the lazy consensus @eladkal sent today at the devlist covers the case - once we agree to the proposal there, i think we are free to release next wave of providers with 3.8 -3.11 support

File: CI.rst (hunk 1 above)
Comment (potiuk) [reply CI.rst]: solved.

**Comment by potiuk** [comment]
Marked as "Ready for review" - should be such and I guess we can merged it soon after @eladkal's lazy consensus.

The doubt was raised whether the providers should be 3.8 only as well - I believe so, it's the same reason we speed up the removal in CI, so providers should be 3.8+ as well IMHO.

**Comment by eladkal** [comment]
> The doubt was raised whether the providers should be 3.8 only as well - I believe so

They should be 3.8+

**Comment by potiuk** [comment]
So - it's all green, ready to final review and merge :)

**Review by uranusjr — APPROVED** [review]

**Comment by pankajkoti** [comment]
Just wish to check if we need to tag this to some milestone and if need to also apply the `pinned` label like in PR https://github.com/apache/airflow/pull/27264?

**Comment by potiuk** [comment]
> Just wish to check if we need to tag this to some milestone and if need to also apply the `pinned` label like in PR #27264?

No need. We are not cherry-picking it to 2.6 (so by default it goes to next major relase). And `pinned` was really there to make sure the 3.11 PR is not closed automatically by the stalebot (the 3.11 was opened in October and it has long period of waiting for dependencies to catch up, so I wanted to make sure stalebot will not close it. The "pinned" label is specifically for stalebot.

**Comment by raphaelauv** [comment]
README of branch main is no more showing 3.7 as a valide version for 2.6.1

Screenshot from 2023-05-29 12-53-21

Is that voluntary ?

**Comment by potiuk** [comment]
> README of branch main is no more showing 3.7 as a valide version for 2.6.1
> 
> Screenshot from 2023-05-29 12-53-21
> 
> Is that voluntary ?

Nope. mistake.

**Comment by potiuk** [comment]
Fix in https://github.com/apache/airflow/pull/31592

**Comment by pankajkoti** [comment]
> > Just wish to check if we need to tag this to some milestone and if need to also apply the `pinned` label like in PR #27264?
> 
> No need. We are not cherry-picking it to 2.6 (so by default it goes to next major relase). And `pinned` was really there to make sure the 3.11 PR is not closed automatically by the stalebot (the 3.11 was opened in October and it has long period of waiting for dependencies to catch up, so I wanted to make sure stalebot will not close it. The "pinned" label is specifically for stalebot.

Okay, understood, nice. Checked it here now that how we save such PRs from the Stalebot here: https://github.com/apache/airflow/blob/main/.github/workflows/stale.yml#L40

**Pull request merged**